Source/WebCore/animation/DocumentTimeline.cpp

```diff
@@ -83,6 +83,7 @@ AnimationTimelinesController* DocumentTimeline::controller() const
 
 void DocumentTimeline::detachFromDocument()
 {
+    Ref protectedThis { *this };
```

Comment:
I think smart pointer guidelines say we should be protecting at the call site instead. Can’t we change `auto& timeline` to `Ref timeline` in `AnimationTimelinesController::detachFromDocument`?

Reply:
Fixed.